### ISSUE TITLE
Cannot override Contract Resolver

### ISSUE THREAD

**Issue by maverickcalibre** [thiagosgarcia/JsonTruncateCsharp#1]
Can we override contractresolver to serialize based on passed in contract resolver ?
Currently Contract Resolver is attached to CustomContractResolver instance while serializing and deserializing. But JsonSerializerSettings might have a contract resolver specified and here we are overriding that contract resolver with CustomContractResolver.

**Comment by thiagosgarcia** [comment]
Yeah, this might be useful. The only problem is that `CustomContractResolver` currently is the one that does all the 'truncate magic', and in order to override it, one resolver specified in `JsonSerializerSettings` should be inherited from it.
Alternatively I think I can work on a way to inject the needed method on any resolver with reflection.
Gonna give it a try and return to you later.
